File: 351003/Suchok/discussion/src/main/java/by/liza/discussion/service/ModerationService.java
package by.liza.discussion.service;

import org.springframework.stereotype.Service;
import java.util.Set;

/**
 * Простая модерация на основе стоп-слов.
 * APPROVE если стоп-слов нет, DECLINE иначе.
 */
@Service
public class ModerationService {
    private static final Set<String> STOP_WORDS = Set.of(
            "spam", "hate", "abuse", "violence", "forbidden"
    );

    public String moderate(String content) {
        if (content == null) return "DECLINE";
        String lower = content.toLowerCase();
        for (String word : STOP_WORDS) {
            if (lower.contains(word)) return "DECLINE";
        }
        return "APPROVE";
    }
}

File: 351002/Khmel/Task340/Discussion/Services/ModerationService.cs
namespace Discussion.Services
{
    public interface IModerationService
    {
        string ModerateContent(string content);
    }

    public class ModerationService : IModerationService
    {
        private readonly ILogger<ModerationService> _logger;
        
        private readonly HashSet<string> _stopWords = new(StringComparer.OrdinalIgnoreCase)
        {
            "spam", "bad", "hate", "abuse", "offensive", "inappropriate"
        };

        public ModerationService(ILogger<ModerationService> logger)
        {
            _logger = logger;
        }

        public string ModerateContent(string content)
        {
            if (string.IsNullOrWhiteSpace(content))
            {
                _logger.LogWarning("Empty content received for moderation");
                return "DECLINE";
            }

            var words = content.ToLower().Split(' ', StringSplitOptions.RemoveEmptyEntries);
            
            foreach (var word in words)
            {
                if (_stopWords.Contains(word))
                {
                    _logger.LogWarning("Stop word '{Word}' found in content", word);
                    return "DECLINE";
                }
            }

            if (content.Length < 2 || content.Length > 2048)
            {
                _logger.LogWarning("Content length {Length} is out of range", content.Length);
                return "DECLINE";
            }

            _logger.LogInformation("Content approved by moderation");
            return "APPROVE";
        }
    }
}